File: 2265-partition-array-according-to-given-pivot/partition-array-according-to-given-pivot.go
func pivotArray(nums []int, pivot int) []int {
    var res []int
    var high []int
    var c int;
    for i:=0;i<len(nums);i++ {
        if nums[i]<pivot{ 
            res=append(res,nums[i])
        }
        if nums[i]==pivot { 
            c++
        }
        if nums[i]>pivot { 
            high=append(high,nums[i])
        }
    }
    for ;c>0;c-- {
        res=append(res,pivot)
    }
    res=append(res,high...)
    return res 
}

File: 2265-partition-array-according-to-given-pivot/partition-array-according-to-given-pivot.cpp
class Solution {
public:
    vector<int> pivotArray(vector<int>& nums, int pivot) {
        vector<int> res,more;
        int c=0;
        for(auto i:nums){
            if(i<pivot) res.push_back(i);
            else if(i==pivot) c++;
            else more.push_back(i);
        }
        while(c--) res.push_back(pivot);
        for(int i:more) res.push_back(i);
        return res;
    }
};